### PR TITLE
Framework

### PR DESCRIPTION
**Pull request by AleksandrVoyteshenko**

### CODE REVIEW

**Review by nitrogen16 — CHANGES_REQUESTED** [review]

File: src/test/java/model/VirtualMachine.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package model;
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/java/model/VirtualMachine.java:1]: fixed

File: src/test/java/service/CreatorVM.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    private static final String valueInstance = "testdata.valueInstance";
+    private static final String valueOperatingSystem = "testdata.valueOperatingSystem";
+    private static final String desiredResultVMClass = "testdata.desiredResultVMClass";
+    private static final String valueSeries = "testdata.valueSeries";
+    private static final String desiredResultInstanceType = "testdata.desiredResultInstanceType";
+    private static final String desiredResultLocalSSD = "testdata.desiredResultLocalSSD";
+    private static final String desiredResultRegion = "testdata.desiredResultRegion";
+    private static final String desiredResultCommitmentTerm = "testdata.desiredResultCommitmentTerm";
+
+    public static VirtualMachine withCredentialsFromProperty() {
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/java/service/CreatorVM.java:16]: getVirtualMachineWithDataFromProperties

File: src/test/java/test/HardCoreTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                .selectCommitedUsage(vm.getDesiredResultCommitmentTerm())
+                .clickAddToEstimateButton();
+        String sumInCalculator = googleCalculatorHCPage.getRentalAmountReceivedInCalculator();
+        GeneratingRandomMailPage generatingRandomMailPage = googleCalculatorHCPage.openNewWindow();
+        String mail = generatingRandomMailPage.generatingAndCopyMailAddress(URL_MAIL);
+        googleCalculatorHCPage
+                .openPopUp();
+        EmailYourEstimatePopUpPage emailYourEstimatePage = new EmailYourEstimatePopUpPage(driver);
+        emailYourEstimatePage
+                .addMailAddress(mail, nameValueField);
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/java/test/HardCoreTest.java:59]: adds an email address in the required field and sends a letter

File: src/test/java/driver/DriverSingleton.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package driver;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import java.time.Duration;
+
+public class DriverSingleton {
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/java/driver/DriverSingleton.java]: Fixed

File: src/test/resources/aqa.properties [hunk 5]
```diff
@@ -0,0 +1,8 @@
+testdata.valueInstance = 4
+testdata.valueOperatingSystem = Free: Debian, CentOS, CoreOS, Ubuntu, or other User Provided OS
+testdata.desiredResultVMClass = Regular
+testdata.valueSeries = n1
+testdata.desiredResultInstanceType = n1-standard-8
+testdata.desiredResultLocalSSD = 2x375
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/resources/aqa.properties:6]: test data

File: src/test/resources/dev.properties [hunk 6]
```diff
@@ -0,0 +1,8 @@
+testdata.valueInstance = 1
+testdata.valueOperatingSystem = Free: Debian, CentOS, CoreOS, Ubuntu, or other User Provided OS
+testdata.desiredResultVMClass = Regular
+testdata.valueSeries = n2
+testdata.desiredResultInstanceType = n2-standard-8
+testdata.desiredResultLocalSSD = 1x375
```
[the remark this answers is not in the document]
Comment (AleksandrVoyteshenko) [reply src/test/resources/dev.properties:6]: test data

**Comment by AleksandrVoyteshenko** [comment]
Merge branch framework

**Pull request merged**